Including files that `#include` other files not supported?

### Installed product versions
- Visual Studio: Community 2017 - 15.9.17
- This extension: 0.7.80

For example this setup does not work.

File `a.frag`:
```c
#version 330 core
out vec4 FragColor;

#include "b1.glsl"
#include "b2.glsl"

void main()
{
	FragColor = vec4(1);
}
```

File `b1.glsl`, and `b2.glsl`:

```c
#ifndef B1
#define B1

#include "c.glsl"

float b1func()
{
	return cfunc();
}

#endif
```

```c
#ifndef B2
#define B2

#include "c.glsl"

float b2func()
{
	return cfunc();
}

#endif
```

and file `c.glsl`:
```c
#ifndef C1
#define C1

float cfunc()
{
	return 1.0;
}

#endif
```
I am using [this class](https://github.com/tntmeijs/GLSL-Shader-Includes) to load my shaders. It handles recursive includes and relative file paths.

However this plugin throws the error `0(7) : error C7529: OpenGL does not allow #include directives` whenever you have more than one layer of includes.

--- a/src/Errors/ShaderCompiler.cs
+++ b/src/Errors/ShaderCompiler.cs
@@ -103,12 +103,12 @@ namespace DMS.GLSL.Errors
 
 			string GetIncludeCode(string includeName)
 			{
-				var includeFileName = Path.Combine(shaderFileDir, includeName);
+			    var includeFileName = Path.Combine(shaderFileDir, includeName);
 				if (File.Exists(includeFileName))
 				{
 					var includeCode = File.ReadAllText(includeFileName);
 					includeCode = SpecialCommentReplacement(includeCode, "//!");
-					return includeCode;
+					return ExpandedCode(includeCode, Path.GetDirectoryName(includeFileName));
 				}
 				return $"#error include file '{includeName}' not found";
 			}

--- a/src/Errors/ShaderCompiler.cs
+++ b/src/Errors/ShaderCompiler.cs
@@ -85,8 +85,11 @@ namespace DMS.GLSL.Errors
 			}
 		}
 
-		private static string ExpandedCode(string shaderCode, string shaderFileDir)
+		private static string ExpandedCode(string shaderCode, string shaderFileDir, HashSet<string> includedFiles=null)
 		{
+			if (includedFiles == null)
+				includedFiles = new HashSet<string>();
+
 			string SpecialCommentReplacement(string code, string specialComment)
 			{
 				var lines = code.Split(new[] { '\n' }, StringSplitOptions.None); //if UNIX style line endings still working so do not use Envirnoment.NewLine
@@ -103,12 +106,18 @@ namespace DMS.GLSL.Errors
 
 			string GetIncludeCode(string includeName)
 			{
-				var includeFileName = Path.Combine(shaderFileDir, includeName);
+				var includeFileName = Path.GetFullPath(Path.Combine(shaderFileDir, includeName));
+
 				if (File.Exists(includeFileName))
 				{
 					var includeCode = File.ReadAllText(includeFileName);
 					includeCode = SpecialCommentReplacement(includeCode, "//!");
-					return ExpandedCode(includeCode, Path.GetDirectoryName(includeFileName));
+
+					if (includedFiles.Contains(includeFileName))
+						return includeCode;
+					includedFiles.Add(includeFileName);
+
+					return ExpandedCode(includeCode, Path.GetDirectoryName(includeFileName), includedFiles: includedFiles);
 				}
 				return $"#error include file '{includeName}' not found";
 			}

--- a/src/Errors/ShaderCompiler.cs
+++ b/src/Errors/ShaderCompiler.cs
@@ -103,7 +103,7 @@ namespace DMS.GLSL.Errors
 
 			string GetIncludeCode(string includeName)
 			{
-			    var includeFileName = Path.Combine(shaderFileDir, includeName);
+				var includeFileName = Path.Combine(shaderFileDir, includeName);
 				if (File.Exists(includeFileName))
 				{
 					var includeCode = File.ReadAllText(includeFileName);

